### ISSUE TITLE
[Bug]: triplot with 'ls' argument yields TypeError  

### ISSUE THREAD

**Issue by maweigert** [matplotlib/matplotlib#21995]
### Bug summary

`plt.triplot` currently yields an error when providing the linestyle via the `ls` argument:

```python

plt.triplot([0,2,1],[0,0,1],[[0,1,2]], ls='--')

>>> TypeError
Got both 'ls' and 'linestyle', which are aliases of one another

```

The reason seems to be https://github.com/matplotlib/matplotlib/blob/f6e0ee49c598f59c6e6cf4eefe473e4dc634a58a/lib/matplotlib/tri/triplot.py#L46 leading to both `linestyle` and `ls` being present in the `kwargs` for `Axes.plot`. 

Possible workaround might be to normalize kwargs at the beginning like in `Axes.plot`:

https://github.com/matplotlib/matplotlib/blob/491ef065f15dd37ccae86fa726bd91601d0c5f72/lib/matplotlib/axes/_axes.py#L1633

### Code for reproduction

```python
plt.triplot([0,2,1],[0,0,1],[[0,1,2]], linestyle=':')
```

### Actual outcome

```
>>> TypeError
Got both 'ls' and 'linestyle', which are aliases of one another
```

### Expected outcome

All fine

### Matplotlib Version

3.4.3

**Comment by jklymak** [comment]
That seems like a reasonable fix.  Thanks!

**Comment by tacaswell** [comment]
I labeled this as good first issue because there is no API discussion to be had and there are clear examples of the machinery we use to handle the aliases on other methods.

The work here:

1. add that machinery to `triplot`
2. add a test (probably with `check_figures_equal` and pass in the line style via both `ls` and `linestyle`)

**Comment by gajendra0180** [comment]
Hey! I am working on the issue! I had set it up in the local machine and the task 1 is done

>1 . add that machinery to triplot

And yes it worked . I am attaching the screenshot of the same.
![Screenshot from 2021-12-18 12-44-12](https://user-images.githubusercontent.com/68333209/146632911-ec4b03b1-f566-4a7c-a074-24045dde8110.png)

**Comment by jklymak** [comment]
Please read our contributing guide.  Thanks!

**Comment by gajendra0180** [comment]
I have made the PR for the issue. Please review it . Thanks.

**Comment by gajendra0180** [comment]
#22001

**Comment by tacaswell** [comment]
Closed in #2201
